src/components/DatePicker/DatePicker.tsx

```diff
@@ -10,14 +10,42 @@ import { DatePickerTrigger } from './DatePickerTrigger';
 import { IconCaretLeft, IconCaretLeftDouble, IconCaretRight, IconCaretRightDouble } from '@foundation/Icon';
 import { Validation } from '@utilities/validation';
 
+type SingleDatePickerProps = {
+    variant?: 'single';
+    onChange: (date: Date | null) => void;
+    selectsRange?: false;
+    startDate?: null;
+    endDate?: null;
+};
+
+type RangeDatePickerProps = {
+    variant: 'range';
+    onChange: (date: [Date | null, Date | null] | null) => void;
+    selectsRange: true;
+    startDate: Date | null;
+    endDate: Date | null;
+};
+
 export type DatePickerProps = {
     placeHolder?: string;
     isClearable?: boolean;
     shouldCloseOnSelect?: boolean;
-    onChange?: (date: Date | null) => void;
     dateFormat?: string;
-    value?: Date;
+    value?: Date | null;
```

Comment:
If I understood correctly. `value` is used in the single variant and for range, you get the values from startDate and endDate. To avoid confusion, I would recommend adding a comment about it in the type definition using JsDocs -> `/** @description Some info there */`

Reply:
Would it be clearer to remove `value` from `DatePickerProps` and add `value?: Date` to `SingleDatePickerProps` and `value?: null` to `RangeDatePickerProps`? This would clarify the relationship + make it more type safe

Reply:
Even better.

Reply:
https://reactdatepicker.com/#example-date-range-for-one-datepicker  -  the range variant uses: `startDate`, `endDate` and `selected`(which) we call value. It uses the `selected` to add some css classes, so this typing makes sense for both variants

Reply:
In this case, I think that some JSDocs on the property wouldn't hurt. I would easily get confused switching variant.

Reply:
Done 👍